components/FileUploadInput.tsx

```diff
@@ -0,0 +1,52 @@
+// TODO form action and submit as no-js fallback
+import React, { useRef } from "react";
+
+type Props = {
+  onFileSelected: (fileObj: File) => void
+}
+
+export default function FileUploadInput({ onFileSelected }: Props) {
+  const inputFile = useRef<HTMLInputElement>(null);
+
+  const onOpenFileInput = (e: React.MouseEvent) => {
+    inputFile.current?.click();
+    e.preventDefault();
+  }
+
+  const onHandleKeyPress = (e: React.KeyboardEvent) => {
```

Comment:
Same thing, is this necessary or is it basically just saying "Prevent the default click and then ... click it anyways"?